streamlit-subscribe/subscribe.py

```diff
@@ -0,0 +1,64 @@
+import streamlit as st
+import psycopg
+import altair as alt
+import threading
+import queue
+import pandas as pd
+import os
+
+DATABASE_URL = os.environ['DATABASE_URL']
+updates_queue = queue.Queue()
+
+def fetch_data():
+    conn = psycopg.connect(DATABASE_URL)
+    with conn.cursor() as cur:
+        for row in cur.stream("SUBSCRIBE simple_sensor_summary ENVELOPE UPSERT (KEY (sensor_id));"):
+            # print(f"Row from database: {row}")  # Log the fetched row for debugging
+            updates_queue.put(row)
+
+# Create a background thread to fetch data
+thread = threading.Thread(target=fetch_data)
+thread.start()
+
+# Initialize the session state
+if 'data' not in st.session_state:
+    st.session_state.data = {
```

Comment:
Could this just be a `defaultdict(list)` (https://docs.python.org/3/library/collections.html#defaultdict-examples) so you don't have to initialize all the keys?

Reply:
Neat! Thank you for suggesting that!